internal/peer/node/start.go

```diff
@@ -825,14 +825,18 @@ func serve(args []string) error {
 			peerAdapter := &commit.PeerAdapter{
 				Peer: peerInstance,
 			}
+			notifier := commit.NewNotifier(peerAdapter)
 			gatewayprotos.RegisterGatewayServer(
 				peerServer.Server(),
 				gateway.CreateServer(
 					&gateway.EndorserServerAdapter{Server: serverEndorser},
 					discoveryService,
 					&commit.Finder{
 						Query:    peerAdapter,
-						Notifier: commit.NewNotifier(peerAdapter),
+						Notifier: notifier,
+					},
+					&commit.Eventer{
+						Notifier: notifier,
 					},
```

Comment:
I started with the narrower interface but was required to do it this way by another maintainer. The way the code is currently structured (at the request of another maintainer), we can't pass concrete implementations in to `gateway.CreateServer` since mocks are passed in there by unit tests.

I'll look at some changes back towards my original approach for the `Finder` and `Eventer`, and I propose we look at changes to other areas of the code in a different PR rather than shoe-horn a bunch of unrelated code refactor into this one.

Reply:
I've made some changes to reduce the leakage of internal abstractions outside of the gateway package, which I think is along the lines of what you are asking.

Note that even within the gateway packages there are deliberately different layers of abstraction. The core Gateway code is intended to be agnostic to whether it is an embedded or standalone implementation, and so interacts with the embedded implementation-specific elements using interfaces, which could be replaced standalone implementations.

Reply:
We can definitely do it in a separate PR. But I feel that there are way to many layers of abstraction for a simple enough functionality.

Reply:
> we can't pass concrete implementations in to gateway.CreateServer

contrary to this you were actually passing the peer instance to `commit package`. What my suggestion was that only external dependency of gateway package is to get an object on which it can call "RegisterForNotification" for a given channel and that should be easy enough to reflect in first-order via an interface.

Reply:
> What my suggestion was that only external dependency of gateway package is to get an object on which it can call "RegisterForNotification" for a given channel and that should be easy enough to reflect in first-order via an interface.

No.

`RegisterForNotification()` implies the peer embedded implementation, which is in the commit package. The commit package uses the `NotificationSupplier` interface to define `CommitNotifications()` which is the same as `RegisterForNotification()`, so it's already doing as you suggest.

The gateway server is not intended to be tied to either a peer embedded or standalone implementation, so it uses a `CommitFinder` interface to obtain transaction validation codes by some means that it doesn't need to know about, and which would be significantly different in the embedded and standalone implementations.